src/frequenz/client/dispatch/_client.py

```diff
@@ -170,6 +180,48 @@ def to_interval(
             else:
                 break
 
+    def stream(self, microgrid_id: int) -> channels.Receiver[DispatchEvent]:
+        """Receive a stream of dispatch events.
+
+        This function returns a receiver channel that can be used to receive
+        dispatch events.
+        An event is one of [CREATE, UPDATE, DELETE].
+
+        Example usage:
+
+        ```
+        client = Client(key="key", server_url="grpc://fz-0004.frequenz.io")
+        async for message in client.stream(microgrid_id=1):
+            print(message.event, message.dispatch)
+        ```
+
+        Args:
+            microgrid_id: The microgrid_id to receive dispatches for.
+
+        Returns:
+            A receiver channel to receive the stream of dispatch events.
+        """
+        return self._get_stream(microgrid_id).new_receiver()
+
+    def _get_stream(
+        self, microgrid_id: int
+    ) -> GrpcStreamBroadcaster[StreamMicrogridDispatchesResponse, DispatchEvent]:
+        """Get an instance to the streaming helper."""
+        if self.streams.get(microgrid_id) is None:
+            request = StreamMicrogridDispatchesRequest(microgrid_id=microgrid_id)
+            self.streams[microgrid_id] = GrpcStreamBroadcaster(
+                stream_name="StreamMicrogridDispatches",
+                stream_method=lambda: cast(
+                    AsyncIterator[StreamMicrogridDispatchesResponse],
+                    self.stub.StreamMicrogridDispatches(
+                        request, metadata=self._metadata
+                    ),
+                ),
+                transform=DispatchEvent.from_protobuf,
+            )
+
+        return self.streams[microgrid_id]
```

Comment:
Very silly optimization to do only one lookup :stuck_out_tongue: 

```suggestion
        broadcaster = self.streams.get(microgrid_id)
        if broadcaster is None:
            request = StreamMicrogridDispatchesRequest(microgrid_id=microgrid_id)
            broadcaster = GrpcStreamBroadcaster(
                stream_name="StreamMicrogridDispatches",
                stream_method=lambda: cast(
                    AsyncIterator[StreamMicrogridDispatchesResponse],
                    self.stub.StreamMicrogridDispatches(
                        request, metadata=self._metadata
                    ),
                ),
                transform=DispatchEvent.from_protobuf,
            )
            self.streams[microgrid_id] = broadcaster

        return broadcaster
```